CRM/Financial/BAO/Payment.php

```diff
@@ -253,22 +258,23 @@ public static function filterUntestedTemplateVariables($params) {
       'paidBy',
       'isShowLocation',
       'location',
-    ];
-    // Need to do these before switching the form over...
-    $todoParams = [
       'isRefund',
-      'totalPaid',
+      'isAmountzero',
       'refundAmount',
+      'totalPaid',
       'paymentsComplete',
+    ];
+    // These are assigned by the payment form - they still 'get through' from the
+    // form for now without being in here but we should ideally load
+    // and assign. Note we should update the tpl to use {if $billingName}
+    // and ditch contributeMode - although it might need to be deprecated rather than removed.
+    $todoParams = [
       'contributeMode',
-      'isAmountzero',
       'billingName',
       'address',
       'credit_card_type',
       'credit_card_number',
       'credit_card_exp_date',
-      'eventEmail',
-      '$event.participant_role',
```

Comment:
@eileenmcnaughton should we remove these two event parameters (as an indicator)? I didn't found any occurrence in form where its been assigned

Reply:
@monishdeb so that was my list of things to add to meet the tpl's requirements but when I reviewed the tpl I decided they were derived parameters (with the tpl) not assigned parameters

Reply:
ok I see